towhee/functional/mixins/column.py

```diff
@@ -0,0 +1,96 @@
+# Copyright 2021 Zilliz. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import pyarrow as pa
+
+
+class ColumnMixin:
+    """
+    Mixins to support column-based storage.
+    """
+    def col_table(self):
+        """
+        Create a column-based table.
+
+        Examples:
+
+        >>> from towhee import Entity, DataFrame
+        >>> e = [Entity(a=a, b=b) for a,b in zip(['abc', 'def', 'ghi'], [1,2,3])]
+        >>> df = DataFrame(e)
+        >>> table = df.col_table()
+        >>> table
+        pyarrow.Table
+        a: string
+        b: int64
+        ----
+        a: [["abc","def","ghi"]]
+        b: [[1,2,3]]
+        """
+        header = [*self._iterable[0].__dict__]
+        cols = [[] for _ in header]
+
+        # cols = [[getattr(entity, col) for entity in self._iterable] for col in header]
+        def inner(entity):
+            nonlocal cols, table
```

Comment:
unused and undeclared `table`, should it be `header`?

Reply:
I forgot to delete this from previous version.